### PR TITLE
doc: correct comment on VJP non-diff input placeholder (NaN, not zero)

### LINKED ISSUE #189: Comment / code mismatch in VJP non-diff placeholder: comment says "zero", code uses NaN
### Description

At `tesseract_jax/tesseract_compat.py:225-232` the inline comment and the code disagree on what gets returned:

```python
elif (
    tan_idx < len(has_tangent)
    and not is_static_mask[all_idx]
    and not has_tangent[tan_idx]
):
    # Non-differentiable but non-static input: return zero gradient
    # with the same shape/dtype as the corresponding input array
    out.append(
        np.full(
            array_args[array_idx].shape,
            np.nan,
            dtype=array_args[array_idx].dtype,
        )
    )
```

Pick one. The JVP path at line 161 uses `np.nan` deliberately as a sentinel (tested at `tests/test_api.py:225` via `assert jnp.all(jnp.isnan(jvp["metadata"]))`), so if the VJP path means to follow the same convention the comment should say "NaN sentinel" to match. Otherwise change the value to `0`.

This branch's output ends up in a tuple slot that JAX's transpose rule doesn't read for any user-requested derivative (see #188 follow-up), so this is doc/code consistency only, not a behaviour fix.

### Steps to reproduce

n/a — read the source.

### Logs

n/a.

### OS

Linux

### Tesseract + Tesseract-JAX version

tesseract-jax `main` (verified 2026-05-20).

### PR DESCRIPTION
**Pull request by andrinr**
#### Relevant issue or PR

Closes #189.

#### Description of changes

The comment at `tesseract_jax/tesseract_compat.py:225-226` said *"return zero gradient"* while the code fills the slot with `np.nan`:

```python
# Non-differentiable but non-static input: return zero gradient
# with the same shape/dtype as the corresponding input array
out.append(
    np.full(array_args[array_idx].shape, np.nan, dtype=...),
)
```

Update the comment to describe the actual behaviour, add a one-liner about why the slot exists (tuple-length contract; JAX's transpose machinery doesn't consume it for any user-requested derivative — see #188's follow-up comment for the empirical check), and cross-reference the JVP path at line ~157 which uses the same NaN sentinel convention.

No behaviour change.

#### Testing done

Comment-only edit; no runtime behaviour to test. Diff is six lines added, two removed.

### CODE REVIEW

File: tesseract_jax/tesseract_compat.py [hunk 1]
```diff
@@ -222,8 +222,12 @@ def vector_jacobian_product(
...
                 and not has_tangent[tan_idx]
             ):
-                # Non-differentiable but non-static input: return zero gradient
-                # with the same shape/dtype as the corresponding input array
+                # Non-differentiable but non-static input: return a NaN
+                # placeholder of the same shape/dtype as the corresponding
+                # input array. The slot exists for tuple-length contract;
+                # JAX's transpose machinery doesn't consume it for any
+                # user-requested derivative. Mirrors the NaN sentinel used
+                # in the JVP path for non-differentiable outputs (line ~157).
```
Comment (dionhaefner) [inline tesseract_jax/tesseract_compat.py]:
```suggestion
                # Non-differentiable but non-static input: return a NaN
                # placeholder of the same shape/dtype as the corresponding
                # input array. The slot exists for tuple-length contract;
                # JAX's transpose machinery doesn't consume it for any
                # user-requested derivative.
```

**Review by dionhaefner — APPROVED** [review]
